automation/taskcluster/lib/tasks.py

```diff
@@ -8,143 +8,110 @@
 
 
 class TaskBuilder(object):
-    def __init__(self, task_id, repo_url, branch, commit, owner, source, scheduler_id, tasks_priority='lowest'):
+    def __init__(self, task_id, repo_url, branch, commit, owner, source, scheduler_id, build_worker_type, beetmover_worker_type, tasks_priority='lowest'):
         self.task_id = task_id
         self.repo_url = repo_url
         self.branch = branch
         self.commit = commit
         self.owner = owner
         self.source = source
         self.scheduler_id = scheduler_id
+        self.build_worker_type = build_worker_type
+        self.beetmover_worker_type = beetmover_worker_type
         self.tasks_priority = tasks_priority
 
-    def raw_task(self, name, description, command, dependencies=[],
-                   artifacts={}, scopes=[], routes=[], features={},
-                   worker_type='github-worker'):
-        created = datetime.datetime.now()
-        expires = taskcluster.fromNow('1 year')
-        deadline = taskcluster.fromNow('1 day')
+    def craft_build_like_task(
+        self, name, description, command, dependencies=None, artifacts=None, scopes=None,
+        routes=None, features=None
+    ):
+        dependencies = [] if dependencies is None else dependencies
+        artifacts = {} if artifacts is None else artifacts
+        scopes = [] if scopes is None else scopes
+        routes = [] if routes is None else routes
+        features = {} if features is None else features
 
-        return {
-            "workerType": worker_type,
-            "taskGroupId": self.task_id,
-            "schedulerId": self.scheduler_id,
-            "expires": taskcluster.stringDate(expires),
-            "retries": 5,
-            "created": taskcluster.stringDate(created),
-            "tags": {},
-            "priority": self.tasks_priority,
-            "deadline": taskcluster.stringDate(deadline),
-            "dependencies": [self.task_id] + dependencies,
-            "routes": routes,
-            "scopes": scopes,
-            "requires": "all-completed",
-            "payload": {
-                "features": features,
-                "maxRunTime": 7200,
-                "image": "mozillamobile/android-components:1.15",
-                "command": [
-                    "/bin/bash",
-                    "--login",
-                    "-cx",
-                    command
-                ],
-                "artifacts": artifacts,
-            },
-            "provisionerId": "aws-provisioner-v1",
-            "metadata": {
-                "name": name,
-                "description": description,
-                "owner": self.owner,
-                "source": self.source
-            }
+        payload = {
+            "features": features,
+            "maxRunTime": 7200,
+            "image": "mozillamobile/android-components:1.15",
```

Comment:
Unrelated with this PR, I was wondering, we might want to revisit this at some point and rather read this Docker image from a config file or something. So that each time we bump it, we no longer have to `grep` and replace everywhere.